notification/endpoint/slack.go

```diff
@@ -0,0 +1,108 @@
+package endpoint
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/influxdata/influxdb"
+)
+
+var _ influxdb.NotificationEndpoint = &Slack{}
+
+const slackTokenSuffix = "-token"
+
+// Slack is the notification endpoint config of slack.
+type Slack struct {
+	Base
+	// URL is a valid slack webhook URL
+	// TODO(jm): validate this in unmarshaler
+	// example: https://slack.com/api/chat.postMessage
+	URL string `json:"url"`
+	// Token is the bearer token for authorization
+	Token influxdb.SecretField `json:"token"`
```

Comment:
I would strongly prefer not to store secrets in plaintext unless we have a good reason to do so.